src/nanotron/fp8/utils.py

```diff
@@ -0,0 +1,13 @@
+import torch
+import transformer_engine as te  # noqa
+
+from nanotron.fp8.constants import FP8_GPU_NAMES
+
+
+def is_fp8_available() -> bool:
+    """Check if FP8 is available on the current device."""
+    if torch.cuda.is_available():
+        device_name = torch.cuda.get_device_name(torch.cuda.current_device()).lower()
+        return any(gpu_name in device_name for gpu_name in FP8_GPU_NAMES)
```

Comment:
I'm surprised TE doesn't have an util to check this. Did you search for it?

Reply:
sadly nope :) https://github.com/search?q=repo%3ANVIDIA%2FTransformerEngine%20torch.cuda.get_device_name&type=code